### PR TITLE
Put python's logs into Trial when running unit tests

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

File: tests/unittest.py [hunk 1]
```diff
@@ -14,21 +14,30 @@
 # limitations under the License.
 import twisted
 from twisted.trial import unittest
+from synapse.util.logcontext import LoggingContextFilter
+from twisted.logger import Logger, LogLevel
```
Comment (richvdh) [inline tests/unittest.py]: Maybe we should use absolute names for these (`import twisted.logger; tx_log=twisted.logger.Logger()`) to reduce confusion between twisted and stdlib logging infrastructure

File: tests/unittest.py [hunk 2]
```diff
@@ -14,21 +14,30 @@
 # limitations under the License.
 import twisted
 from twisted.trial import unittest
+from synapse.util.logcontext import LoggingContextFilter
```
Comment (richvdh) [inline tests/unittest.py]: ftr, PEP8 has some [opinions](https://www.python.org/dev/peps/pep-0008/#imports) on import order which we try to follow - in this case that means that imports from `synapse` go after those from `twisted`. OTOH this file already breaks the convention by putting `import logging` at the end so \<shrug>

**Comment by richvdh** [comment]
This is all very well, but how do I get the damn logs out when I run the tests?

**Comment by hawkowl** [comment]
_trial_temp/test.log is where Trial puts it by default.

**Comment by richvdh** [comment]
hrm ok, but it seems very sparse. Is there a way to make it do debug logs?

**Comment by richvdh** [comment]
[it seems to do debug logs just fine. I'm not sure what I was smoking on Sunday.]

**Pull request merged**